hazelcast/src/main/java/com/hazelcast/nio/tcp/TcpIpConnectionManager.java

```diff
@@ -530,6 +543,7 @@ private void closeServerSocket() {
     private void stop() {
         live = false;
         log(Level.FINEST, "Stopping ConnectionManager");
+        handlerBalancer.stop();
```

Comment:
If you sleep for 20 seconds, then the thread is not going to terminate itself for 20 seconds. With an interrupt you can let him break free.